docker-app/qfieldcloud/core/management/commands/extractstoragemetadata.py

```diff
@@ -0,0 +1,127 @@
+import csv
+import logging
+import sys
+from pathlib import Path
+from typing import Generator, NamedTuple
+
+import boto3
+import mypy_boto3_s3
+import yaml
+from django.core.management.base import BaseCommand, CommandParser
+from qfieldcloud.core import utils
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Save metadata from S3 storage project files to disk."""
+
+    def add_arguments(self, parser: CommandParser):
+        parser.add_argument("-o", "--output", type=str, help="Name of output file")
+        parser.add_argument(
+            "-f",
+            "--config_file",
+            type=str,
+            help="Path to YAML file with your S3 credentials",
+        )
+
+    def handle(self, *args, **options):
+        config = None
+
+        if options.get("config_file"):
+            path = Path(options["config_file"])
+
+            if not path.is_file():
+                logger.error(f"This path does not exist or is not a file: {path}")
+                sys.exit(1)
+
+            config = self.from_file(path)
+            bucket = self.get_s3_bucket(config)
+
+        else:
+            bucket = utils.get_s3_bucket()
+
+        output_name = options.get("output")
+
+        if output_name:
+            name_suffix = output_name.rsplit(".", 1)
+
+            if len(name_suffix) != 2 or name_suffix[1] != "csv":
+                logger.error("The name of the output file should end with '.csv'")
+                sys.exit(1)
+        else:
+            output_name = "s3_storage_files.csv"
+
+        fields = ["id", "key", "e_tag", "size", "create_time"]
+        rows = self.read_bucket_files(bucket, fields)
+
+        with open(output_name, "w") as fh:
+            writer = csv.writer(fh, delimiter=",")
+            writer.writerow(fields)
+            writer.writerows(rows)
+
+        logger.info(f"Successfully exported data to {output_name}")
+
+    class S3ConfigObject(NamedTuple):
+        STORAGE_ACCESS_KEY_ID: str
+        STORAGE_SECRET_ACCESS_KEY: str
+        STORAGE_BUCKET_NAME: str
+        STORAGE_REGION_NAME: str
+        STORAGE_ENDPOINT_URL: str
+
+    @staticmethod
+    def from_file(path_to_config: Path) -> S3ConfigObject:
+        """Load a YAML file exposing credentials used by S3 storage"""
+        with open(path_to_config) as fh:
+            contents = yaml.safe_load(fh)
+
+        config = {
+            k: v for k, v in contents.items() if k in Command.S3ConfigObject._fields
+        }
```

Comment:
Can you rewrite this as proper `for` with `if` inside, in the qfc codebase inlining is acceptable for very simple cases?